### PR TITLE
LPS-166565: Add mappings for text embeddings on startup to avoid need for full re-indexing

### PR DESCRIPTION
**Pull request by peerkar**
https://issues.liferay.com/browse/LPS-166565

### CODE REVIEW

**Comment by peerkar** [comment]
ci:test:relevant

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/ml/embedding/instance/lifecycle/EmbeddingFieldCreationPortalInstanceLifecycleListener.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+/**
+ * @author Petteri Karttunen
+ */
+@Component(enabled = false, service = PortalInstanceLifecycleListener.class)
+public class EmbeddingFieldCreationPortalInstanceLifecycleListener
+	extends BasePortalInstanceLifecycleListener {
+
+	@Override
+	public void portalInstanceRegistered(Company company) throws Exception {
```
Comment (lipusz) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/ml/embedding/instance/lifecycle/EmbeddingFieldCreationPortalInstanceLifecycleListener.java:47]: Is this triggered on each startup when the company instance is initialized (so if I move from Update X to Update Y) or only when a Company instance is being created the first time (so when a new Virtual Instance is added)?

**Comment by peerkar** [comment]
ci:forward

**Comment by BryanEngler** [comment]
Just wondering if we should be using IndexSettingsContributor? Is there anything that guarantees that EmbeddingFieldCreationPortalInstanceLifecycleListener runs before SearchIndexPortalInstanceLifecycleListener? If not I think the put mapping will fail if the mapping doesnt exist?

**Comment by peerkar** [comment]
@BryanEngler thanks, I'll add the missing indexExists check there, so that is run only only on existing index - if those templates are missing. 

I'm not sure about using IndexSettingsContributor because they are run on index creation / re-indexing only? The point of this was to handle adding those field on existing index. As far as I see, that kind of scenario would however only materialize on product upgrade, upgrading from version without semantic search to version with it. But as you asked in the ticket: do we force re-index anyways? In that case this would be not necessary at all? @lipusz ?

**Comment by BryanEngler** [comment]
i think we are fine if the put mapping fails when then index doesnt exist, because the mappings are already added to liferay-type-mappings.json, so once the index is created, the mappings will be there.

IndexSettingsContributor would be the correct path to take if the mappings were not present in liferay-type-mappings.json

**Comment by peerkar** [comment]
ci:test:relevant

**Comment by peerkar** [comment]
ci:reevaluate:1377401_8275

**Comment by peerkar** [comment]
ci:test:relevant

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
ci:reevaluate:1357401_3419

**Comment by joshchong** [comment]
ci:test:search

**Comment by peerkar** [comment]
Forwarding here https://github.com/brianchandotcom/liferay-portal/pull/129282

**Pull request closed without merging**